### PR TITLE
[Loss] Split logging loss from backward loss

### PR DESCRIPTION
**Pull request by HAOCHENYE**
**This PR (2/5)** — behavior-neutral. Routes displayed/logged losses through detached per-rank local components reduced by cross-rank **SUM**; the backward loss is still globally reduced. Isolates the display pipeline so the next PR flips gradients without touching logging.

Full stack (merge bottom-up, under #1959):
1. `reduce-sum-01-fsdp-helper` — [FSDP] Add reduce-sum gradient reduction helper
2. `reduce-sum-02-split-logging` — [Loss] Split logging loss from backward loss
3. `reduce-sum-03-switch-sum` — [FSDP][Loss] Switch gradient reduction to SUM
4. `reduce-sum-04-remove-world-size` — [Refactor] Remove unused world_size plumbing
5. `reduce-sum-05-drop-nonglobal` — [Refactor] Drop non-global aux-loss averaging mode

### CODE REVIEW

File: xtuner/v1/engine/train_engine.py [hunk 1]
```diff
@@ -240,12 +243,14 @@ def train_step(self, data_batches: list[ModelItem]) -> TrainStepInfo:
 
             loss = self._get_total_loss(output)
             loss.backward()
-            total_loss += loss.detach()
+            local_display_loss += self.model.local_display_loss(output)
```
Comment (HAOCHENYE) [inline xtuner/v1/engine/train_engine.py:246]: loss reduction, otherwise still do it in post_micro_batch_forward?

File: xtuner/v1/loss/aux_loss.py [hunk 2]
```diff
@@ -169,15 +184,22 @@ def finalize(
...
         if z_list:
             partials = [ctx.finalize() for ctx in z_list]
-            z_loss = partials[0] if len(partials) == 1 else torch.stack(partials).sum(dim=0)
+            z_losses = [p[0] for p in partials]
+            z_locals = [p[1] for p in partials]
+            z_loss = z_losses[0] if len(z_losses) == 1 else torch.stack(z_losses).sum(dim=0)
+            local_z_loss = z_locals[0] if len(z_locals) == 1 else torch.stack(z_locals).sum(dim=0)
 
-        return balancing_loss, z_loss, tokens_per_expert_global
+        return balancing_loss, z_loss, tokens_per_expert_global, local_balancing_loss, local_z_loss
```
Comment (HAOCHENYE) [inline xtuner/v1/loss/aux_loss.py:202]: Return a TypedDict, as the tuple return type is not convenient for extension

File: xtuner/v1/engine/train_engine.py [hunk 3]
```diff
@@ -240,12 +243,14 @@ def train_step(self, data_batches: list[ModelItem]) -> TrainStepInfo:
...
             # call dump_forward_records after backward to record the recomputed activations
             ProberList.after_micro_iter_forward()
 
+        if dist.is_initialized():
+            dist.all_reduce(local_display_loss, op=dist.ReduceOp.SUM)
```
Comment (HAOCHENYE) [inline xtuner/v1/engine/train_engine.py:251]: This part of the logic is placed in post_micro_batch_forward, and I hope each rank can display its own loss without performing an all_reduce sum. Could you try multiplying this coefficient back instead of doing a global sum?

File: xtuner/v1/loss/moe_loss.py [hunk 4]
```diff
@@ -139,34 +139,49 @@ def finalize(
...
+            loss_vec = scale_global * (tokens_per_expert_local * routing_weights_mean_global).sum(-1)
+            # No cross-rank term in this branch; the loss is already a per-rank quantity.
+            local_vec = loss_vec.detach()
 
-        loss = scale_global * (tokens_per_expert_for_loss * routing_weights_mean_global).sum(-1)
-        loss = loss.sum() * self.loss_cfg.balancing_loss_alpha
-        return loss / self._batch_size
+        loss = loss_vec.sum() * alpha / self._batch_size
+        local_loss = local_vec.sum() * alpha / self._batch_size
+        return loss, local_loss.detach()
```
Comment (HAOCHENYE) [inline xtuner/v1/loss/moe_loss.py:184]: Why return two losses? If a subsequent module needs to record it, just detach it there. Why implement it this way? The interfaces between modules are all very strange now

File: xtuner/v1/model/moe/moe.py [hunk 5]
```diff
@@ -657,12 +686,13 @@ def _micro_batch_forward(
             z_ctx=z_ctx,
             non_pad_token=non_pad_token,
         )
-        balancing_loss, z_loss, tokens_per_expert_global = split_aux_output
+        balancing_loss, z_loss, tokens_per_expert_global, local_balancing_loss, local_z_loss = split_aux_output
```
Comment (HAOCHENYE) [inline xtuner/v1/model/moe/moe.py:689]: If local_loss can be omitted, please simplify the interface

File: xtuner/v1/engine/train_engine.py
Comment (HAOCHENYE) [inline xtuner/v1/engine/train_engine.py:1]: I don't see the need to make so many changes just for printing loss

**Comment by HAOCHENYE** [comment]
Superseded by the consolidated 2-PR stack #1968 (core) + #1969 (drop non-global). Closing this fragmented stack.
